backend/capellacollab/sessions/exceptions.py

```diff
@@ -33,8 +53,72 @@ async def unsupported_session_type_handler(
     )
 
 
+async def conflicting_session_handler(
+    request: fastapi.Request, exc: ConflictingSessionError
+) -> fastapi.Response:
+    return await exception_handlers.http_exception_handler(
+        request,
+        fastapi.HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail={
+                "err_code": "EXISTING_SESSION",
+                "reason": (
+                    f"You already have a '{exc.tool.name}' session with version '{exc.version.name}'. "
+                    "Please terminate the existing session or reconnect to it."
+                ),
+            },
+        ),
+    )
+
+
+async def tool_and_model_mismatch_handler(
+    request: fastapi.Request, exc: ToolAndModelMismatchError
+) -> fastapi.Response:
+    return await exception_handlers.http_exception_handler(
+        request,
+        fastapi.HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail={
+                "err_code": "MODEL_VERSION_MISMATCH",
+                "reason": (
+                    f"The model '{exc.model.name}' is not compatible with the tool {exc.version.tool.name} ({exc.version.name})'."
+                ),
+            },
+        ),
+    )
+
+
+async def invalid_connection_method_identifier_handler(
+    request: fastapi.Request, exc: InvalidConnectionMethodIdentifierError
+) -> fastapi.Response:
+    return await exception_handlers.http_exception_handler(
+        request,
+        fastapi.HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
```

Comment:
I changed it to `HTTP_422_UNPROCESSABLE_ENTITY` already. `status.HTTP_404_NOT_FOUND` doesn't match here since the resource itself is found, but not a sub-resource passed via the body.

> In addition, I would either remove the ({exc.version.name}) in the exception above or add it here to be consistent.

The connection method is defined on the tool itself, so no version reference is needed. In the exception above it's about a conflicting version, so we should display the version.